template/logentry/template.proto

```diff
@@ -33,4 +33,10 @@ message Template {
 
     // Severity indicates the importance of the log entry.
     string severity = 3;
+
+    // An expression to compute the type of the monitored resource this log entry is being recorded on.
+    string monitored_resource_type = 4;
```

Comment:
string is too limited, e.g. Google service control uses labels, map[string]string to identify resource. 
istio.mixer.v1.config.descriptor.ValueType is more flexible. How about calling the field monitored_resource, so it doesn't have to be a resource type?